src/fides/api/ops/service/connectors/saas_connector.py

```diff
@@ -116,7 +116,16 @@ def retrieve_data(
 
         query_config: SaaSQueryConfig = self.query_config(node)
         read_request: Optional[SaaSRequest] = query_config.get_request_by_action("read")
+        delete_request: Optional[SaaSRequest] = query_config.get_request_by_action(
+            "delete"
+        )
+
         if not read_request:
+            # if a delete request is specified for this endpoint without a read request
+            # then we return a single empty row to still trigger the mask_data method
+            if delete_request:
+                return [{}]
```

Comment:
As you've highlighted here, some requests don't need to interact on row data at all, but we have this current constraint where erasures are tightly tied to the rows from access requests. The access is run first to generate "rows" of data, which are passed to erasures and we often then generate a masking query or request _per_ row.  This design makes more sense for databases, but for saas connectors it is not always necessary. 

Maybe a future direction would involve this checking if its delete_request "needs_rows"(pretend variable name) is False then it just returns instead of a fake row. Then when the masking request runs, if its "needs_rows" is False, it doesn't have to pretend to loop through rows, it just makes its singular request.   Some GDPR delete endpoints might benefit from this, if the access request generated five rows, but we just want to make a singular GDPR delete , not five.

But there's so many places that loop through rows right now, I think this is a good way to unblock this functionality in a way that works within the current constraints.